src/services/rulehelper.js

```diff
@@ -379,7 +379,7 @@ ngeo.RuleHelper = class {
    */
   createFilter(options) {
 
-    const dataSource = options.dataSource;
+    const dataSource = /** @type {ngeo.DataSource} */ (options.dataSource);
```

Comment:
Unfortunately there is no way to check that: we can not do an assert on a typedef.